### PR TITLE
Use a higher value for connecion_pool_maxsize.

### LINKED ISSUE #26: Consider setting connection_pool_maxsize to a higher value by default
Currently, it is set to 1. Which means the moment you start doing concurrent requests to the same kubernetes master, you're going to end up thrashing the connection pool & reducing performance.

Since the most likely case is that we're going to be making many requests to the same IP rather than many requests to different IP, setting this to a larger value by default will (and does!) improve performance.

### PR DESCRIPTION
**Pull request by pokoli**
Fixes #26

### CODE REVIEW

**Comment by mbohlool** [comment]
This looks good to me. I will let @yuvipanda test and confirm this before merging. Thanks @pokoli.

File: configuration.py [hunk 1]
```diff
@@ -83,11 +84,11 @@ def __init__(self):
...
         # urllib3 connection pool's maximum number of connections saved
-        # per pool. Increasing this is useful for cases when you are
-        # making a lot of possibly parallel requests to the same host,
-        # which is often the case here.
-        # When set to `None`, will default to whatever urllib3 uses
-        self.connection_pool_maxsize = None
+        # per pool. urllib3 uses 1 connection as default value, but this is
+        # not the best value when you are making a lot of possibly parallel
+        # requests to the same host, which is often the case here.
+        # cpu_count * 5 is used as default value to increase performance
```
Comment (yuvipanda) [inline configuration.py]: Add a note saying we're doing this because this is the default for ThreadPoolExecutor?

Comment (pokoli) [inline configuration.py]: Added, thanks.

**Comment by pokoli** [comment]
Thanks for the review guys!

**Pull request merged**